linux/shim/io.go

```diff
@@ -15,8 +15,9 @@ import (
 )
 
 func copyConsole(ctx context.Context, console console.Console, stdin, stdout, stderr string, wg, cwg *sync.WaitGroup) error {
+	fctx := context.Background()
 	if stdin != "" {
-		in, err := fifo.OpenFifo(ctx, stdin, syscall.O_RDONLY, 0)
+		in, err := fifo.OpenFifo(fctx, stdin, syscall.O_RDONLY, 0)
```

Comment:
The client currently opens the fifos with NONBLOCK and do so before the create call is performed.